src/databricks/labs/ucx/source_code/jobs.py

```diff
@@ -441,3 +458,76 @@ def _process_dependency(
         linter = FileLinter(ctx, path_lookup, self._session_state, dependency.path, inherited_tree)
         for advice in linter.lint():
             yield LocatedAdvice(advice, dependency.path)
+
+
+class DfsaCollectorWalker(DependencyGraphWalker[DirectFsAccess]):
+
+    def __init__(
+        self,
+        graph: DependencyGraph,
+        walked_paths: set[Path],
+        path_lookup: PathLookup,
+        session_state: CurrentSessionState,
+    ):
+        super().__init__(graph, walked_paths, path_lookup)
+        self._session_state = session_state
+
+    def _process_dependency(
+        self, dependency: Dependency, path_lookup: PathLookup, inherited_tree: Tree | None
+    ) -> Iterable[DirectFsAccess]:
+        language = file_language(dependency.path)
+        if not language:
+            logger.warning(f"Unknown language for {dependency.path}")
+            return
+        cell_language = CellLanguage.of_language(language)
+        source = dependency.path.read_text(guess_encoding(dependency.path))
+        if is_a_notebook(dependency.path):
+            yield from self._collect_from_notebook(source, cell_language, dependency.path, inherited_tree)
+        elif dependency.path.is_file():
+            yield from self._collect_from_source(source, cell_language, dependency.path, inherited_tree)
+
+    def _collect_from_notebook(
+        self, source: str, language: CellLanguage, path: Path, inherited_tree: Tree | None
+    ) -> Iterable[DirectFsAccess]:
+        notebook = Notebook.parse(path, source, language.language)
+        for cell in notebook.cells:
+            src_timestamp = int(path.stat().st_mtime)
+            src_id = str(path)
+            src_lineage = self.lineage_str
+            for dfsa in self._collect_from_source(cell.original_code, cell.language, path, inherited_tree):
+                yield dfsa.replace(
+                    source_type="NOTEBOOK", source_id=src_id, source_lineage=src_lineage, source_timestamp=src_timestamp
```

Comment:
```suggestion
                    source_type="NOTEBOOK", source_id=src_id, source_lineage=src_lineage, source_timestamp=src_timestamp,
```
it reformats to multiple lines then

Reply:
this code is not the latest